### PR TITLE
Add base64 favicon and social sharing meta tags

### PR DESCRIPTION
**Pull request by jessicaschilling**
Small enhancements to dir-index-html pages: 
- Closes https://github.com/ipfs/dir-index-html/issues/35 - not only prevents 404 error on an implicit /favicon.ico but also gives the tab an IPFS cube favicon consistent with other IPFS sites.
 - Closes https://github.com/ipfs/dir-index-html/issues/34 - adds social sharing metadata. Proof of display in comments below.

### CODE REVIEW

**Comment by jessicaschilling** [comment]
Thanks to @lidel for raising the idea of an inline favicon! Sending to both @Stebalien and @aschmahmann for review, since I'm not sure whose purview this falls under these days.

image

**Comment by Stebalien** [comment]
@aschmahmann is big boss now.

**Comment by jessicaschilling** [comment]
Proof of display for social sharing cards ... note that these are all using https://gateway.ipfs.io/ipfs/QmUj1ZWPLTvzZhzUFqmthsCwWm6Tn8rF5cyV9h8xCSKVQ2 as a test, which is simply a view-source of my local instance of this branch saved as regular HTML.

Slack: 
image

Twitter (via https://cards-dev.twitter.com/validator):
image

Facebook (via https://developers.facebook.com/tools/debug/):
image

File: dir-index-uncat.html [hunk 1]
```diff
@@ -2,6 +2,21 @@
...
+  <meta property="og:type" content="website">
+  <meta property="og:image" content="https://gateway.ipfs.io/ipfs/QmSDeYAe9mga6NdTozAZuyGL3Q1XjsLtvX28XFxJH8oPjq">
+  <meta name="twitter:title" content="Files on IPFS">
+  <meta name="twitter:description" content="A directory of files hosted on the distributed, decentralized web using IPFS">
+  <meta name="twitter:card" content="summary_large_image">
+  <meta name="twitter:image" content="https://gateway.ipfs.io/ipfs/QmSDeYAe9mga6NdTozAZuyGL3Q1XjsLtvX28XFxJH8oPjq">
+  <meta name="twitter:creator" content="@ipfs">
+  <meta name="twitter:site" content="@ipfs">
+  <meta name="image" content="https://gateway.ipfs.io/ipfs/QmSDeYAe9mga6NdTozAZuyGL3Q1XjsLtvX28XFxJH8oPjq">
+  <meta name="keywords" content="IPFS, dweb, protocol, libp2p, ipld, multiformats, bitswap, decentralized web, InterPlanetary File System, dapp, Protocol Labs">
```
Comment (lidel) [inline dir-index-uncat.html:17]: I worry these keywords will do more harm than good: as google bots crawl public gateways and see it everywhere, seacher engines will quickly learn to ignore them and perhaps even mark them as "spammy words" and decrease SEO juice. I'd say remove it (better safe than sorry).

```suggestion
```

File: dir-index-uncat.html [hunk 2]
```diff
@@ -2,6 +2,21 @@
 <html>
 <head>
   <meta charset="utf-8" />
+  <meta name="description" content="A directory of files hosted on the distributed, decentralized web using IPFS">
+  <meta property="og:title" content="Files on IPFS">
```
[the remark this answers is not in the document]
Comment (jessicaschilling) [reply dir-index-uncat.html:6]: Nice idea!

File: dir-index-uncat.html (hunk 1 above)
Comment (jessicaschilling) [reply dir-index-uncat.html:17]: Fair enough. Those are the same keywords we are using in IPFS docs, fwiw, but at the sort of scale we might be seeing for dir-index-html instantiations, that could potentially be a concern.

**Comment by jessicaschilling** [comment]
@lidel had a great idea to dynamically insert title element for cases in which someone might be social-previewing multiple links. Amended, and new previews follow using https://gateway.ipfs.io/ipfs/QmNjia5FrrhnrUgtKmzFKvbkJNvAL2Rmc9xxYiVw5GA3xs (again, saved-as regular HTML from my local) ...

Slack:
image

Twitter (via https://cards-dev.twitter.com/validator):
image

Facebook (via https://developers.facebook.com/tools/debug/):
image

@lidel -- you'll see in the Facebook preview that something as long as a CID in that dynamic title field pushes the subtitle out of the available space. I'd actually lean toward restoring the original Facebook meta tags to result in the earlier screenshot -- I know that's not dynamic, and doesn't help differentiate if someone posts a bunch of links to FB, but at least the generic one has some meaning. LMK.

@aschmahmann -- in any case, I think we're close enough to done for you to be able to review this 😊

**Comment by jessicaschilling** [comment]
@lidel OR -- this is actually preferable -- let's just reverse the title and description tags for Facebook (and others who use the `og:foo` meta tags). Semantically it still makes as much sense ... 

Using https://gateway.ipfs.io/ipfs/QmXFhdYS5EwEVp3Ri2jARj9bmmzT1doZk5CzqVrJX2LRsp ...

Slack reverses the order from my previous comment's screenshot, since it uses the same meta tags as FB:
image

Twitter is the same as before:
image

Facebook is better:
image

**Comment by jessicaschilling** [comment]
@lidel -- we chatted briefly about this but just want to confirm you're OK with this as it stands. Thanks!

**Comment by jessicaschilling** [comment]
FYI -- this one is PR'd into go-ipfs as https://github.com/ipfs/go-ipfs/pull/7511

**Pull request merged**